src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractAddItemCommandHandler.TemplateDetails.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
+{
+    internal partial class AbstractAddItemCommandHandler
+    {
+        internal class TemplateDetails
+        {
+            public string CapabilityCheck;
+            public Guid DirNamePackageGuid;
```

Comment:
All these fields can be readonly.

Reply:
Hah, this is my sense of neatness taking over. The two Guids are passed by ref so can't be properties or readonly fields, so I just made them all the same. I'll make the other three properties :)

Reply:
From a correctness point of view these values should never change, I think I would copy them to a field before passing them to a method.

Reply:
Good idea, solves both problems.